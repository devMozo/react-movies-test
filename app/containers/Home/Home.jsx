import React from 'react';
import CategoryItem from '../../components/CategoryItem/CategoryItem';
import { connect } from 'react-redux';
import { getAll } from '../../redux-app/actions/CategoryAction';
import './Home.scss';
import HandlerLoaders from '../../components/HandlerLoaders/HandlerLoaders';

export class Home extends React.PureComponent {
    getAll = () => {
        const { getAll } = this.props;
        getAll();
    };

    componentDidMount() {
        this.getAll();
    }

    render() {
        const { categories, loading, error } = this.props;

        return (
            <main className="Home">
                <HandlerLoaders loading={loading} error={error} onTryAgain={this.getAll}>
                    <ul className="Home__list">
                        {categories &&
                            categories.map(category => (
                                <CategoryItem key={category.id} id={category.id} name={category.name} />
                            ))}
                    </ul>
                </HandlerLoaders>
            </main>
        );
    }
}

const mapStateToProps = state => {
    return {
        ...state.category,
    };
};

const mapDispatchToProps = dispatch => {
    return {
        getAll: () => {
            dispatch(getAll());
        },
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Home);
