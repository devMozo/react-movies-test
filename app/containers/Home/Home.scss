.Home {
    * {
        font-family: 'Arial';
    }

    .Home__list {
        padding: 30px 20px 0 20px;
        display: flex;
        flex-wrap: wrap;
        justify-content: space-between;
    }
}
