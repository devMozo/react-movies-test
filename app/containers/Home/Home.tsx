import * as React from 'react';
import './Home.scss';

export default class Home extends React.PureComponent {
    render() {
        return <></>;
    }
}
