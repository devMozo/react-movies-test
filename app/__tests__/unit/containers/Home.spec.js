import React from 'react';
import { shallow, mount } from 'enzyme';
import { Home } from '../../../containers/Home/Home';
import { BrowserRouter as Router } from 'react-router-dom';

const MOCK_CATEGORIES = require('../../../__mocks__/categories.json');

describe('<Home />', () => {
    let PROPS;
    const mockedGetAll = jest.fn();

    beforeEach(() => {
        PROPS = {
            categories: [],
            loading: false,
            error: false,
            getAll: mockedGetAll,
        };
    });

    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Home {...PROPS} />);
        expect(component).toMatchSnapshot();
    });

    it('should render all categories when the data was fetched', () => {
        PROPS = {
            ...PROPS,
            categories: MOCK_CATEGORIES.genres,
        };

        const component = mount(
            <Router>
                <Home {...PROPS} />
            </Router>,
        );

        expect(component.find('.CategoryItem').length).toBe(MOCK_CATEGORIES.genres.length);
        expect(mockedGetAll).toHaveBeenCalled();
    });
});
