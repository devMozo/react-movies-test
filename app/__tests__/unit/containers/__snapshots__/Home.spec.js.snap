// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly in "debug" mode 1`] = `
<main
  className="Home"
>
  <Memo(HandlerLoaders)
    error={false}
    loading={false}
    onTryAgain={[Function]}
  >
    <ul
      className="Home__list"
    />
  </Memo(HandlerLoaders)>
</main>
`;
